src/main/java/seedu/address/logic/commands/AttendanceAddCommand.java

```diff
@@ -0,0 +1,106 @@
+
+/**
+ * Adds attendance to student in address book.
+ */
+public class AttendanceAddCommand extends Command {
+    public static final String COMMAND_WORD = "add";
+    public static final String MESSAGE_ADD_ATTENDANCE_SUCCESS = "Added attendance to Student: %1$s";
+
+    public static final String MESSAGE_USAGE = ATTENDANCE_COMMAND_WORD + " "
+            + COMMAND_WORD
+            + ": Adds attendance to the student identified "
+            + "by the index number used in the last student listing and size(between 1-12), "
+            + "if size is 0, attendance list will be empty."
+            + "Existing attendance will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_CLASS_GROUP + "[CLASS] "
+            + PREFIX_SIZE + "[LIST SIZE]\n"
+            + "Example: " + ATTENDANCE_COMMAND_WORD
+            + " " + COMMAND_WORD + " 1 "
+            + PREFIX_CLASS_GROUP + "CS2103T " + PREFIX_SIZE + "10";
+
+    private final Index index;
+    private final String mod;
+    private final String size;
+
+    /**
+     * Creates an AttendanceAddCommand to add to the specified {@code Student}
+     * @param index of the person in the filtered person list to edit the remark
+     * @param mod belonging to the attendance list
+     * @param size of the attendance list
+     */
+    public AttendanceAddCommand(Index index, String mod, String size) {
+        requireAllNonNull(index, mod, size);
+        this.index = index;
+        this.mod = mod;
+        this.size = size;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Student> lastShownList = model.getFilteredStudentList();
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        Student studentToEdit = lastShownList.get(index.getZeroBased());
+        if (!AttendanceList.isValidSize(size)) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ATTENDANCE_LIST_INDEX);
+        }
+
+        AttendanceList attendanceList = new AttendanceList(mod, size);
+        Student editedStudent = new Student(
+                studentToEdit.getName(), studentToEdit.getPhone(), studentToEdit.getEmail(),
+                studentToEdit.getClassGroup(), studentToEdit.getStudentId(), studentToEdit.getTags(),
+                attendanceList);
+
+        model.setStudent(studentToEdit, editedStudent);
+        model.updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
+
+        return new CommandResult(generateSuccessMessage(editedStudent));
+    }
+
+    /**
+     * Generates a command execution success message based on whether
+     * the attendance is added to or removed from
+     * {@code studentToEdit}.
+     */
+    private String generateSuccessMessage(Student studentToEdit) {
+        String message = MESSAGE_ADD_ATTENDANCE_SUCCESS;
```

Comment:
Need not create a message variable imo. Just whack the whole MESSAGE_ADD_ATTENDANCE_SUCCESS in.